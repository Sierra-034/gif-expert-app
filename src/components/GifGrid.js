import React from "react";

export const GifGrid = ({ category }) => {

    const getGifs = async () => {
        const URL = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&limit=10&api_key=4N0WEqRJUE4MR2NNALGC8qIYVvuNUXwM';
        const response = await fetch(URL);
        const {data} = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        
        console.log(gifs);
    };

    getGifs();
    
    return (
        <div>
            <h3>{ category }</h3>
        </div>
    );
};