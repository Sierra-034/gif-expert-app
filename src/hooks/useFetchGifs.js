import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    const imgSetter = (imgs) => {
        setTimeout(() => {
            setState({
                data: imgs,
                loading: false
            });
        }, 500);
    };

    useEffect(() => {
        getGifs(category)
            .then(imgSetter);
    }, [category]);

    return state;
};
