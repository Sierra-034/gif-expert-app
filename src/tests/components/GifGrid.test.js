import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre <GifGrid />', () => {

    const category = 'Halo';
    
    test('Coincide con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Muestra los items cuando se cargan imÃ¡genes con useFetchGifs', () => {
        const imgData = [
            {
                id: 'image_id',
                title: 'img_title',
                url: 'http://localhost/some/img.jpg'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: imgData,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
});
