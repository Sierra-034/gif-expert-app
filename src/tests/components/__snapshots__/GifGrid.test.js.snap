// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGrid /> Coincide con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Halo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
